book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/vendor/repository/VendorRepository.java

```diff
@@ -0,0 +1,24 @@
+package org.lbcc.bms.bms_monolith.vendor.repository;
+
+import org.lbcc.bms.bms_monolith.common.entity.Vendor;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Repository
+public interface VendorRepository extends JpaRepository<Vendor, UUID> {
+
+    //TODO: need to add indexing for name based search
+    @Query("SELECT v FROM Vendor v WHERE v.name = :vendorName")
```

Comment:
@Query is not required. findByName is enough. 

change List<Vendor> to Page<Vendor> use pagination and sorting wherever a bulk data is received.